pennylane_lightning/core/src/simulators/lightning_qubit/catalyst/LightningSimulator.cpp

```diff
@@ -22,8 +22,29 @@
 namespace Catalyst::Runtime::Simulator {
 
 auto LightningSimulator::AllocateQubit() -> QubitIdType {
```

Comment:
@maliasadi Are we keeping this functionality in then, rather than only allowing a single allocation & deallocation?

Reply:
There aren't any lowering to these function calls. These methods can be removed from the API if the dynamic qubit management will not be added in a foreseeable future.